corehq/apps/cloudcare/tests/test_api.py

```diff
@@ -1,81 +1,74 @@
-from mock import patch
 import json
 import uuid
 
-from django.test import TestCase
 from django.core.urlresolvers import reverse
+from django.test import TestCase
+from django.test.utils import override_settings
+from mock import patch
+
 from casexml.apps.case.mock import CaseBlock
 from casexml.apps.case.models import CommCareCase
 from casexml.apps.case.util import post_case_blocks
-from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
-from corehq.form_processor.tests.utils import FormProcessorTestUtils, run_with_all_backends
-from toggle.shortcuts import update_toggle_cache, clear_toggle_cache
 from corehq import toggles
+from corehq.apps.cloudcare.api import get_filtered_cases, CaseAPIResult, CASE_STATUS_OPEN, CASE_STATUS_ALL,\
+    CASE_STATUS_CLOSED
+from corehq.apps.cloudcare.views import ReadableQuestions
 from corehq.apps.domain.shortcuts import create_domain
 from corehq.apps.users.models import CommCareUser
 from corehq.apps.users.util import format_username
-from corehq.apps.cloudcare.views import ReadableQuestions
-from corehq.apps.cloudcare.api import get_filtered_cases, CaseAPIResult, CASE_STATUS_OPEN, CASE_STATUS_ALL,\
-    CASE_STATUS_CLOSED
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.form_processor.tests.utils import FormProcessorTestUtils
+from toggle.shortcuts import update_toggle_cache, clear_toggle_cache
 
 TEST_DOMAIN = "test-cloudcare-domain"
 
 
-class CaseAPITest(TestCase):
-    """
-    Tests some of the Case API functions
-    """
+class CaseAPIBaseTest(object):
     domain = TEST_DOMAIN
     case_types = ['t1', 't2']
     user_ids = ['TEST_API1', 'TEST_API2']
 
     @classmethod
-    def setUpClass(cls):
-        super(CaseAPITest, cls).setUpClass()
+    def do_setup(cls):
         create_domain(cls.domain)
         cls.password = "****"
 
         def create_user(username):
+            username = format_username(username, cls.domain)
+            ccu = CommCareUser.get_by_username(username)
+            if ccu:
+                ccu.delete()
```

Comment:
Sorry, wasn't meant to leave that in there. Was just an artifact of the test failing in setup and using reuse db. I'll clean it up